### PR TITLE
Stop remediation happening on policies where the assignment_enforcement_mode is set to false.

### PR DESCRIPTION
**Pull request by pmatthews05**
## Description
- When a assignment (initiative or definition) is set, and the `assignment_enforcement_mode` is set to false, it shouldn't apply a remediation. An extra condition to check for this assignment_enforcement_mode.
- For data calls to `azurerm_policy_set_definition` change calls to lookup from name, as Microsoft changes display names, update examples, and readme files.
- Switch `data.azurerm_policy_definition` to `data.azurerm_policy_definition_built_in` and using the name instead of the display name. Updated examples and readme files.
- Updated azurerm version number to `>=3.44.0` in examples and readme files, because `data.azurerm_policy_definition_built_in` is only available from that version onwards.
- Formatted the tf files, using newly created script .\format-terraform.ps1

Fixes # (issue)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

**Test Configuration**:
* Module Version: Latest
* Terraform Version: 1.4.4 on Windows_amd64
* AzureRM Provider Version: 3.5.0

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by gettek** [comment]
Hi @pmatthews05, thank you for raising this, great spot about preventing remediation when `assignment_enforcement_mode=false`.

Could you kindly update your branch with the changes made in #73, have finally caved into running `terraform fmt` on all modules since noticing this PR, have also modified the existing pre-commit script so we can remove `format-terraform.ps1`

**Comment by pmatthews05** [comment]
@gettek I have pulled your code into mine using a separate branch. Went through all the conflicts. I then merged this branch into my main.
That should make this PR correct now, with no merge conflicts and has both #73 and #74.

**Comment by pmatthews05** [comment]
@gettek I see you have added some tags and assigned to me. Is there anything else you require for me to do?

**Comment by gettek** [comment]
@pmatthews05 looking forward to merge this but made a few comments above if you wouldn't mind resolving first please?

**Comment by gettek** [comment]
@pmatthews05 if you scroll through the [files changes](https://github.com/gettek/terraform-azurerm-policy-as-code/pull/72/files) (or above) I have made a few more comments for minor changes 🙂

**Comment by pmatthews05** [comment]
@gettek I'm sorry, I've looked at file changes and there are no comments from you for minor changes. Do you mind re-listing them in a comment back to me please?

**Review by gettek — CHANGES_REQUESTED** [review]

**Comment by gettek** [comment]
@pmatthews05 apologies, did forgot to hit the submit review button!

File: modules/def_assignment/main.tf [hunk 1]
```diff
@@ -80,6 +80,7 @@ resource "azurerm_subscription_policy_assignment" "def" {
   }
 }
 
+
```
Comment (gettek) [inline modules/def_assignment/main.tf:83]: Please remove this extra line

**Review by gettek — CHANGES_REQUESTED** [review]
Please fix the final comment and run precommit to update the readme's back to the old tf versioning, thanks!

**Pull request merged**